File: 98.validate_binary_search_tree/isValidBST.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isValidBST = function(root) {
    if (root === null) {
        return true;
    }
    var arr = []
    inOrder(root, arr)
    for(var i = 0; i < arr.length-1; i++) {
        if (arr[i] >= arr[i+1]) {
            return false;
        }
    }
    return true;
};

var inOrder = function(root, arr) {
    if (root.left !== null) {
        inOrder(root.left, arr);
    }
    arr.push(root.val)
    if (root.right !== null) {
        inOrder(root.right, arr);
    }
}

File: 98.validate_binary_search_tree/isValidBST.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isValidBST(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        output = []
        self.inOrder(root, output)
        for i in range(1, len(output)):
            if output[i] <= output[i-1]:
                return False
        return True

    def inOrder(self, parent, output):
        if not parent:
            return
        self.inOrder(parent.left, output)
        output.append(parent.val)
        self.inOrder(parent.right, output)
